### DISCUSSION TITLE
Gotenberg on Cloud-run throws "[FATAL] starting libreoffice-api: launch supervisor: start process: LibreOffice socket not available: context deadline exceeded"

### DISCUSSION THREAD

**Discussion by bahag-xvisnei** [gotenberg/gotenberg#745] [Q&A] [answered]
Hi, I have deployed gotenberg:7.10.1-cloudrun on my dev GCP project with these configurations:

CMD ["gotenberg", "--api-port=8080", "--chromium-disable-routes=true", "--webhook-disable=true", "--prometheus-disable-route-logging=true", "--pdfengines-disable-routes=true", "--libreoffice-restart-after=3", "--libreoffice-auto-start=true", "--log-level=debug"]

Right after a fresh deployment, it works fine. For my dev GCP project, however, I don't need it to run 7/24. So it scales down to 0 instances. Once it scales down to 0 (neither active nor idle instance), upon receiving a new request afterwards, gotenberg throws the error "LibreOffice socket not available: context deadline exceeded". Below is a screenshot of my cloud-run logs:

![image](https://github.com/gotenberg/gotenberg/assets/104824015/d4098904-ac60-42fb-8ebe-313dc0c111cf)

I configured the health check settings so it waits for readiness:

![image](https://github.com/gotenberg/gotenberg/assets/104824015/fe7401da-e750-4f42-a669-d9d142222023)

I run the instance with 1GB Ram and 1 CPU. My word files are simple and 1-2 pages only.

do you have some suggestion on how to make it run stable?

**Comment by gulien** [comment, marked as answer]
Hello @bahag-xvisnei,

You may try to increase the startup timeout with `--libreoffice-start-timeout=30s` 🤔

**Reply by bahag-xvisnei** [reply]
Hi @gulien 

Thanks for the answer.

I have tried multiple combinations, I understand now better.

Cloud run, if configured, can test the readiness of the container. Startup probe can test the TCP connection to determine service's readiness.

Our container starts first listening on the TCP port before the Libreoffice is started fully as can be seen in the logs below:

2023-12-12 08:29:38.625 CET [SYSTEM] api: server listening on port 8080
2023-12-12 08:29:48.554 CET [SYSTEM] libreoffice-api: LibreOffice automatically started

It would be much easier to control the readiness of the container, if the container would start listening only after all routes become fully initialized.

Does it makes sense to you too? Can we make a change in future releases?

Bests

**Reply by gulien** [reply]
That may be doable 👍

**Reply by gulien** [reply]
Added it to my TODO list #750.

**Reply by bahag-xvisnei** [reply]
Great!

**Reply by gulien** [reply]
@bahag-xvisnei `gotenberg/gotenberg:edge-cloudrun` is currently building but should soon be available (~1h). The API now waits for all modules to be ready before starting listening on the configured port.

**Reply by bahag-xvisnei** [reply]
@gulien This is great. You are fast 👍 

I will test it and let you know.

**Reply by bahag-xvisnei** [reply]
Hi @gulien 

I have tested it and it seems working. Great work 👍 

I set the initial delay to 5 seconds(in my experience it needs at least ~10 seconds) and set the retry count to 10 times with 1 second delay in between. After 4 attempts GCP could test the TCP connection successfully and did proceed with the request.

![image](https://github.com/gotenberg/gotenberg/assets/104824015/22b417b8-6702-4984-95f0-7792cb89bb4c)

**Reply by gulien** [reply]
Great, thanks!
